api/utils/addr.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"net"
+	"net/url"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+// ExtractHostPort takes addresses like "tcp://host:port/path" and returns "host:port".
```

Comment:
I'm sure* (well, _sort-of_ sure) I've seen copies of this sort URL-part extraction in a few different places. Is it worth consolidating them?

Reply:
Yeah, this logic was moved from `lib/utils/addr.go` with simplifications and other changes. I don't know if it's similar enough to be reused in `lib/utils/addr.go` at this point, but I'll take a crack at it to avoid duplication.

Reply:
I'm going to instead move the `Extract` functions to local functions for webclient. `lib/utils/addr.go` should be the go to utils package for address parsing, but since that can't be used within the `api` module I think it's best to keep the footprint small and add a few helper local functions.